numpy/f2py/cfuncs.py

```diff
@@ -1007,14 +1007,16 @@
     PyObject* tmp = NULL;
     if (PyFloat_Check(obj)) {
         *v = PyFloat_AsDouble(obj);
-        return !(*v == -1.0 && PyErr_Occurred());
+        if (*v == -1.0 && PyErr_Occurred()) { goto error; }
+        return 1;
```

Comment:
Test this by adding an object with `__float__()` which has an error and which does not have an error.

Reply:
Generally test this a bit more (`cmocka`?)